src/Test/Constraints/XmlMatchesXsdConstraint.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Test\Constraints;
+
+/**
+ * @author SpacePossum
+ *
+ * @internal
+ */
+final class XmlMatchesXsdConstraint extends \PHPUnit_Framework_Constraint
+{
+    /**
+     * @var string[]
+     */
+    private $xmlConstraintErrors = array();
+
+    /**
+     * @var string
+     */
+    private $xsd;
+
+    /**
+     * @param string $xsd
+     */
+    public function __construct($xsd)
+    {
+        parent::__construct();
+
+        // replace first only
+        $needle = 'http://www.w3.org/2001/xml.xsd';
+        if (false !== $pos = strpos($xsd, $needle)) {
+            $xsd = substr_replace($xsd, 'file://'.str_replace('\\', '/', __DIR__).'/xml.xsd', $pos, strlen($needle));
```

Comment:
Why do you need to replace this URI?

Reply:
Technically you don't.... as long as you are online and the resource is available because if you're not than the xsd cannot be pulled and thing breaks.

Reply:
maybe it's worth to document it btw